plugins/woocommerce/src/Internal/DependencyManagement/ExtendedContainer.php

```diff
@@ -15,6 +15,8 @@
 /**
  * This class extends the original League's Container object by adding some functionality
  * that we need for WooCommerce.
+ *
+ * NOTE: This class will be removed in WooCommerce 10.0.
```

Comment:
> This might have limited visibility to any third-party extending or modifying it

This class is inside the `Internal` namespace, this means that [nobody outside we the core maintainers is expected to be using it](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/README.md). The ability to still use it is provided only as a temporary fallback mechanism in case issues are found in the new container. This comment is more a reminder for ourselves.

> was it considered shipping this as disabled experimental, with an option to enable on WC > Settings > features?

This is not really a "feature" in active development but an atomic change in how we organize things internally. Providing it as a feature that can be toggled via UI (and worse, disabled by default) could give the false impression that this is an optional thing and slow down adoption.

Reply:
> nobody outside we the core maintainers is expected to be using it

That said: it's probably a good idea to post a developer advisory about this in the Woo developers blog.